components/bl-particles-component/src/helpers.js

```diff
@@ -0,0 +1,191 @@
+  loadPathShape, loadPerlinNoisePath, loadPolygonMaskPlugin, loadPolygonPath, loadRoundedRectShape,
+  loadSimplexNoisePath, loadSoundsPlugin, loadSpiralShape,
+} from './lib/particles.min';
+import {
+  BigCircles, Bubbles, Colors, Confetti, Direction, Fire, Firefly, Fireworks, Growing,
+  Hyperspace, Light, Links, SeaAnemone, Shapes, Snow, Stars, TetradicConfetti, Triangles,
+} from './options';
+
+const Presets = {
+  BIG_CIRCLES      : 'bigCircles',
+  BUBBLES          : 'bubbles',
+  CONFETTI         : 'confetti',
+  FIRE             : 'fire',
+  FIREFLY          : 'firefly',
+  FIREWORKS        : 'fireworks',
+  GROWING          : 'growing',
+  HYPERSPACE       : 'hyperspace',
+  LIGHT            : 'light',
+  LINKS            : 'links',
+  SEA_ANEMONE      : 'seaAnemone',
+  SNOW             : 'snow',
+  STARS            : 'stars',
+  TETRADIC_CONFETTI: 'tetradicConfetti',
+  TRIANGLES        : 'triangles',
+};
+
+const PresetsMap = {
+  [Presets.BIG_CIRCLES]      : BigCircles,
+  [Presets.BUBBLES]          : Bubbles,
+  [Presets.CONFETTI]         : Confetti,
+  [Presets.FIRE]             : Fire,
+  [Presets.FIREFLY]          : Firefly,
+  [Presets.FIREWORKS]        : Fireworks,
+  [Presets.GROWING]          : Growing,
+  [Presets.HYPERSPACE]       : Hyperspace,
+  [Presets.LIGHT]            : Light,
+  [Presets.LINKS]            : Links,
+  [Presets.SEA_ANEMONE]      : SeaAnemone,
+  [Presets.SNOW]             : Snow,
+  [Presets.STARS]            : Stars,
+  [Presets.TETRADIC_CONFETTI]: TetradicConfetti,
+  [Presets.TRIANGLES]        : Triangles,
+};
+
+const UNSET_COLLISION_MODE = 'unset';
+const LINKS_DISTANCE = 150;
+const STROKE_WIDTH = 2;
+const COLOR_ANIMATION_SPEED = 20;
+const OPACITY = { FULL: 1, LOW: 0.1 };
+const DEFAULT_Z_INDEX = 0;
+const DEFAULT_TIMING = 0;
+const RotateDegrees = { MIN: 0, MAX: 360 };
+const SpiralShapeProps = {
+  particles  : { stroke: { width: STROKE_WIDTH } },
+  innerRadius: 1,
+  lineSpacing: 1,
+  fill       : false,
+  close      : false,
+};
+
+const LineShapeProps = {
+  particles: {
+    stroke: { width: STROKE_WIDTH },
+    move  : { straight: true },
+    rotate: { path: true },
+  },
+};
+
+export function useOptions(component) {
+  const {
+    preset, textValue, color, backgroundColor, autoPlay, fullScreen, shape, zIndex, lineLinksVisibility, move, size,
+    speed, opacity, direction, colorAnimation, number, imageURL, rotate, rotationAnimation, customOptions, linksColor,
+    rollingAnimation, duration, delay, triangleLinksVisibility, collisionMode, outModes, backgroundOpacity,
+  } = component;
+
+  const particles = useMemo(() => ({
+    collisions: { enable: collisionMode !== UNSET_COLLISION_MODE, mode: collisionMode },
+    links     : {
+      enable   : lineLinksVisibility || triangleLinksVisibility,
+      distance : LINKS_DISTANCE,
+      color    : linksColor || Colors.RANDOM,
+      opacity  : OPACITY[lineLinksVisibility ? 'FULL' : 'LOW'],
```

Comment:
`lineLinksVisibility ? OPACITY.FULL : OPACITY.LOW`

same for others

We use this maps for autocomplete, your way for using it miss the sense of it.

Also `OPACITY` -> `Opacity`